Description: gss apparently crashes on fade actions on LDM_DIRECTX exported displays. Since fading is a bad idea anyway on LTSP clients it's disabled completely now if LTSP_CLIENT is set.
Author: Oliver Grawert

Index: gnome-screensaver-2.30.2/src/gs-fade.c
===================================================================
--- gnome-screensaver-2.30.2.orig/src/gs-fade.c	2010-11-15 16:38:09.000000000 +1100
+++ gnome-screensaver-2.30.2/src/gs-fade.c	2010-11-15 16:39:17.746665011 +1100
@@ -410,6 +410,10 @@
         screen_priv = &fade->priv->screen_priv[screen_idx];
 
 #ifdef HAVE_XF86VMODE_GAMMA
+	if (g_getenv("LTSP_CLIENT")) {
+		return FADE_TYPE_NONE;  /* We're on an LTSP Client, bad idea to fade at all */
+	}
+
         res = XF86VidModeQueryExtension (GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()), &event, &error);
         if (! res)
                 goto fade_none;
